TurboMinerVirtualTable.Unity/Assets/Scripts/Ui/Lobby/PlayerManager.cs

```diff
@@ -0,0 +1,98 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+using System.Linq;
+using TMPro;
+
+public class PlayerManager : MonoBehaviour
+{
+    public GameObject PlayersList;
+    public Button StartButton;
+
+    private List<PlayerLabel> PlayersLabels;
+
+    private void Start()
+    {
+        PlayersLabels = new List<PlayerLabel>();
+    }
+
+    private void Update()
+    {
+        if(PlayersLabels.Count > 1 && 
+           PlayersLabels.Select(l => l.GetComponentInChildren<ColorPicker>().image.color).Distinct().Count() == PlayersLabels.Count &&
+           PlayersLabels.Select(l => l.NameLabel.text).Distinct().Count() == PlayersLabels.Count)
+        {
+            StartButton.interactable = true;
+        }
+        else
+        {
+            StartButton.interactable = false;
+        }
+    }
+
+    public void AddPlayer()
+    {
+        if(PlayersLabels.Count < 4)
+        {
+            CreateUi();
+        }
+    }
+
+    private void CreateUi()
```

Comment:
Ta nazwa tez dobra xD
Spoko jest slowo Setup do takich rzeczy
SetupLabels, tez dziwi mnie troche podejscie ze za kazdym dodaniem to musi powstawac na nowo (wg mnie zjebane ale moze czegos nie rozumiem)